src/App.js

```diff
@@ -1,25 +1,78 @@
-import React from 'react';
-import logo from './logo.svg';
+import React, { useState } from 'react';
 import './App.css';
+import contacts from './contacts.json';
 
 function App() {
+  //Iteration 1: State with only the first 5
+ const fiveContacts = contacts.slice(0,5);
+ const [contactsState, setContacts] = useState (fiveContacts);
+
+//Iteration 2
+function addRandomContacts(){
+const randomContact = Math.floor(Math.random() * contacts.length)
+  contactsState.push(contacts.[randomContact]);
+  setContacts(current => [...current, contactsState]);
+}
+
+//Iteration 3 - sort by Name
```

Comment:
One of the main rules of React is that you never want to directly change the state, only using `this.setState()` or in this case using `setContacts()`.

It looks like you are changing `contactsState` directly on line 19 and then updating it on line 28, it works but it's not the recommended way to do it.

Here's my revised version using the same logic
```
function sortName(){
  const contactsStateCopy = [...contactsState];
  contactsStateCopy.sort((a,b) => {
    if (a.name < b.name) {
      return -1;
    }
    if (a.name > b.name) {
      return 1;
    }
    return 0;
  })
  setContacts(contactsStateCopy);
}